packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -784,14 +786,28 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
     }
 
     /**
-     * Stop the current timer for clearing the blob cache (if any) and schedule a new one
+     * Schedule a timer for clearing the blob cache or defer the current one.
      */
     private scheduleClearBlobsCache() {
         const blobCacheTimeoutDuration = 10000;
+        // When the timer runs out, decide whether to proceed with the cache clear or reset the timer
+        const clearCacheOrDefer = () => {
```

Comment:
I think you can you move it into else block and not create this object on every read